Astro 2 upgrade gives Typescript errors re. missing MDX properties: rawContent and compiledContent

### What version of `astro` are you using?

2.0.2

### Are you using an SSR adapter? If so, which one?

none

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

After upgrading my project to latest Astro 2.0.2 and upgrading all the `@astrojs/*` plugins, also `@astrojs/mdx`, running `tsc --noEmit` results in the following missing properties:

```text
tsc --noEmit
node_modules/.pnpm/astro@2.0.2_@types+node@18.11.18/node_modules/astro/client-base.d.ts:
115:31 - error TS2339: Property 'rawContent' does not exist on type 'MDX'.

115  export const rawContent: MDX['rawContent'];
                                  ~~~~~~~~~~~~
node_modules/.pnpm/astro@2.0.2_@types+node@18.11.18/node_modules/astro/client-base.d.ts:
116:36 - error TS2339: Property 'compiledContent' does not exist on type 'MDX'.

116  export const compiledContent: MDX['compiledContent'];
                                       ~~~~~~~~~~~~~~~~~
Found 2 errors in the same file, starting at: node_modules/.pnpm/astro@2.0.2_@types+node@18.11.18/node_modules/astro/client-base.d.ts:115
```

The errors refer to `astro/client-base.d.ts` which has the following code:

<img width="565" alt="astro-2-mdx-error" src="https://user-images.githubusercontent.com/7058116/215312016-38260215-0371-413e-a9fc-e0d4c3dfbf0b.png">

And indeed, these two properties are omitted from `MDXInstance` in `astro.d.ts`:

```ts
export type MDXInstance<T extends Record<string, any>> = Omit<MarkdownInstance<T>, 'rawContent' | 'compiledContent'>;
```

However, the project itself runs fine with `astro dev` command.

### Link to Minimal Reproducible Example

n/a

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/gentle-weeks-lie.md
+++ b/.changeset/gentle-weeks-lie.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Remove `rawContent()` and `compiledContent()` from MDX import types

--- a/packages/astro/client-base.d.ts
+++ b/packages/astro/client-base.d.ts
@@ -112,8 +112,6 @@ declare module '*.mdx' {
 	export const url: MDX['url'];
 	export const getHeadings: MDX['getHeadings'];
 	export const Content: MDX['Content'];
-	export const rawContent: MDX['rawContent'];
-	export const compiledContent: MDX['compiledContent'];
 
 	const load: MDX['default'];
 	export default load;

